src/libraries/System.Linq/src/System/Linq/First.cs

```diff
@@ -69,11 +69,15 @@ public static TSource FirstOrDefault<TSource>(this IEnumerable<TSource> source,
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
             }
 
-            if (source is IPartition<TSource> partition)
-            {
-                return partition.TryGetFirst(out found);
-            }
+            return
+#if !OPTIMIZE_FOR_SIZE
```

Comment:
I also don't love the double-negative quality, but I did it this way in case we were to accidentally not set the flag, in which case we'd silently be missing the throughput optimizations. My hope is also that the flag can entirely go away, but that will require more work / measurement.